crates/api_common/src/community.rs

```diff
@@ -90,6 +91,27 @@ pub struct ListCommunitiesResponse {
   pub communities: Vec<CommunityView>,
 }
 
+
+#[skip_serializing_none]
+#[derive(Debug, Serialize, Deserialize, Clone, Default)]
+#[cfg_attr(feature = "full", derive(TS))]
+#[cfg_attr(feature = "full", ts(export))]
+/// Exports a list of communities to a CSV file.
+pub struct ExportCommunities {
+  pub type_: Option<ListingType>,
+  pub sort: Option<ExportSortType>,
+  pub show_nsfw: Option<bool>,
+  pub auth: Option<Sensitive<String>>,
+}
```

Comment:
This endpoint should go under something like `/api/v3/user/export`, so that other info can be added later. There is no need to include any sorting or filtering options.